tests/alive-tv/memory/memcmp-opt-littleendian-fail.src.ll

```diff
@@ -0,0 +1,21 @@
+target datalayout = "e-p:64:64:64"
+; TEST-ARGS: -disable-undef-input
+
+; ex) %x = 0x1020, %y = 0x30
+; Note that %x > %y
+define i32 @f1(i64 %x, i64 %y) {
+  %p = alloca i64
+  %q = alloca i64
+  store i64 %x, i64* %p, align 1 ; 20 10 00 00 ..
+  store i64 %y, i64* %q, align 1 ; 30 00 00 00 ..
+  %p8 = bitcast i64* %p to i8*
+  %q8 = bitcast i64* %q to i8*
+  %res = call i32 @memcmp(i8* %p8, i8* %q8, i64 8) ; %res < 0
+  ret i32 %res
+}
+
+; ERROR: Value mismatch
```

Comment:
I'm lost with this one. Why is this transformation wrong?

Reply:
This test shows that it is illegal in little-endian to transform memcmp to integer comparison. The reason is that the least significant byte is positioned at the first byte.
For example, if %x is 0x1020 and %y is 0x30, %x > %y, so the target returns positive number, whereas source returns negative number because 0x20 < 0x30.